sql/analyzer/validate_create_table.go

```diff
@@ -367,6 +377,17 @@ func validateAlterIndex(initialSch, sch sql.Schema, ai *plan.AlterIndex, indexes
 	return indexes, nil
 }
 
+// validateIndexType prevents indexing blob columns
+func validateIndexType(cols []sql.IndexColumn, sch sql.Schema) error {
+	for _, c := range cols {
+		i := sch.IndexOfColName(c.Name)
+		if sql.IsBlob(sch[i].Type) {
```

Comment:
This isn't matching the right types and your check is too restrictive as a result

```go
func IsBlob(t Type) bool {
	switch t.Type() {
	case sqltypes.Binary, sqltypes.VarBinary, sqltypes.Blob:
		return true
	default:
		return false
	}
}
```

You'll need to see what other places this is used, but I'm guessing it should be really be called IsByteType or similar